generators/client/templates/angular/src/main/webapp/app/core/util/operators.ts.ejs

```diff
@@ -0,0 +1,22 @@
+<%#
+ Copyright 2013-2021 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+
+export function isPresent<T>(t: T | undefined | null | void): t is T {
+  return t !== undefined && t !== null;
+}
```

Comment:
It seems to me that this adds unnecessary complexity. Simple check can be used instead of this added `isPresent`.

Reply:
I tried too now and after playing a bit my conclusion is that your solution is a good solution. I couldn't do better. So marking this and related comments as resolved.